agent/lib/broker_controller.js

```diff
@@ -248,6 +281,17 @@ function is_not_internal(conn) {
     return conn.user !== undefined && conn.user.match('^agent.[a-z0-9]+$') == null; //Can't get properties or anything else on which to base decision yet
 }
 
+function compare_address_settings(orig_address_settings, new_address_settings) {
+    for (var name in orig_address_settings) {
+        log.debug('comparing %s: %s %s', name, orig_address_settings[name], new_address_settings[name])
+        if (new_address_settings[name] && myutils.string_compare(orig_address_settings[name], new_address_settings[name]) !=0){
```

Comment:
recompilation of string_compare in the false case.

Reply:
updated